### PR TITLE
Using `only printing` and fixing coercion in notations

### LINKED ISSUE #310: clean up using printing only notations
Grep for:
- strings interrupted by `'  '`
- strings starting with `\` a string and  then a blank and the continuation of string

Examples:
- in `mathcomp/ssreflect/finset.v`
  + `Notation "[ 'se' 't' E | x 'in' A , y 'in' B ]"`
  + `Notation "[ 'se' 't' E | x 'in' A , y 'in' B & P ]"`
  + `Notation "[ 'se' 't' E | x : T , y : U 'in' B ]"`
  + `Notation "[ 'se' 't' E | x : T , y : U 'in' B & P ]"`
  + `Notation "[ 'se' 't' E | x : T 'in' A , y : U ]"`
  + `Notation "[ 'se' 't' E | x : T 'in' A , y : U & P ]"`
  + `Notation "[ 'se' 't' E | x : T , y : U ]"`
  + `Notation "[ 'se' 't' E | x : T , y : U & P ]"`]
- in  `mathcomp/ssreflect/fintype.v`
  + `Notation "[ 'pic' 'k' x : T ]" := [pick x : T | pick_true _]`
- in `mathcomp/ssreflect/eqtype.v`
  + `Notation "[ 'sub' 'Type' 'for' v ]" := (SubType _ v _ _ vrefl_rect)`
  + `Notation "[ 'new' 'Type' 'for' v ]" := (NewType v _ _ vrefl_rect)`

### PR DESCRIPTION
**Pull request by CohenCyril**
##### Motivation for this change

Fixes #310 and a printing bug that went under the radar possibly several(?) versions of Coq before.

##### Things done/to do

- [x] added corresponding entries in `CHANGELOG_UNRELEASED.md` (do not edit former entries)
- ~added corresponding documentation in the headers~

##### Automatic note to reviewers

Read [this Checklist](https://github.com/math-comp/math-comp/wiki/Checklist-for-following,-reviewing-and-playing-with-a-PR#checklist-for-reviewing-a-pr) and make sure there is a milestone.

### CODE REVIEW

**Comment by CohenCyril** [comment]
@gares, how do I enable checking the output of the `Check` in files `mathcomp/test_suite/imset2_finset.v` and `mathcomp/test_suite/imset2_gproduct.v` vs `mathcomp/test_suite/imset2_finset.v.out` and `mathcomp/test_suite/imset2_gproduct.v.out`?

**Comment by CohenCyril** [comment]
@gares this looks like a regression in coq dev...

**Comment by CohenCyril** [comment]
@erikmd what is the difference between coq-dev and coq-latest?

**Comment by gares** [comment]
@erikmd it would be nice to have a job testing v8.13 (the branch).
I did merge many notation related PRs, so regressions may have happened there.

**Comment by CohenCyril** [comment]
@gares indeed! broken in 8.12 and fixed in dev!!! good news!!

**Comment by CohenCyril** [comment]
Should I make a special `coq.8.12` test file that is deliberatly broken, to make the CI pass? or should we keep it broken until it is solved on Coq side?

**Comment by CohenCyril** [comment]
Meanwhile, I will replace the output test here not to make use of `{pred T}`, since it was not the point here, open a PR to test `{pred T}` specifically. Is it possible to do so on coq side too?

**Comment by CohenCyril** [comment]
@gares ready to merge if you want

**Comment by CohenCyril** [comment]
@gares this is the last missing piece for releasing tomorrow.

File: CHANGELOG_UNRELEASED.md [hunk 1]
```diff
@@ -507,6 +507,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
   + `coprimez_mul(l|r)` -> `coprimezM(l|r)`
   + `coprimez_exp(l|r)` -> `coprimezX(l|r)`
 
+- in `finset.v`, fixed printing of notation `[set E | x in A , y in B & P ]`
```
Comment (CohenCyril) [inline CHANGELOG_UNRELEASED.md:510]: @gares this changelog entry is in the wrong section, but please disregard, I will curate it (cf. #677)

**Pull request merged**